lib/webdriveragent.js

```diff
@@ -244,223 +236,104 @@ class WebDriverAgent {
-      // at which time the system will not be able to figure out that the process
-      // has started
-      logs.on('output', (stdout, stderr) => {
-        let errorString = 'Could not start logger for udid';
-        if (stdout.indexOf(errorString) !== -1 || stderr.indexOf(errorString) !== -1) {
-          // unfortunately we have no way to stopping the process, so just log overtly
-          let msg = `The real device logger '${this.realDeviceLogger}' was ` +
-                    `unable to start log capture. Please try installing ` +
-                    `'deviceconsole' ('npm install -g deviceconsole') and ` +
-                    `specify the path to it using the 'realDeviceLogger' capability.`;
-          log.error(msg);
-        }
-      });
-    }
-
-    if (!this.iosLogAlreadyShown) {
-      logs.on('output', (stdout, stderr) => {
-        let out = stdout || stderr;
-        // make sure we are not reading logs from before this test run
-        if (!loggingStarted && !shouldStartLogging(out)) {
-          return;
-        }
-        if (isPertinentLogLine(out)) {
-          for (let line of out.split("\n").filter(Boolean)) {
-            agentLog.debug(`${prefix}: ${line}`);
-          }
-        }
-      });
-    }
-  }
-
-  async getStartTime () {
-    let startTime = new Date();
-    if (this.realDevice) {
-      let {stdout} = await exec('idevicedate', ['-u', this.device.udid]);
-      startTime = new Date(stdout);
-      if (isNaN(startTime.getTime())) {
-        log.debug('Unable to get device time. Using local system time');
-        // there is never any stderr output, just stdout
-        log.debug(`Output from idevicedate: ${stdout}`);
-        startTime = new Date();
-      }
-    }
-    return startTime;
-  }
-
   async startXcodebuild () {
     // wrap the start procedure in a promise so that we can catch, and report,
     // any startup errors that are thrown as events
-    return await new B(async (resolve, reject) => {
+    return await new B((resolve, reject) => {
       this.xcodebuild.on('exit', (code, signal) => {
         log.info(`xcodebuild exited with code '${code}' and signal '${signal}'`);
+        this.xcodebuild.processExited = true;
         if (this.xcodebuild._wda_error_occurred || (!signal && code !== 0)) {
           return reject(new Error(`xcodebuild failed with code ${code}`));
         }
       });
 
-      this.deviceLogs.on('exit', (code) => {
-        let msg = `${this.realDevice ? 'System' : 'Simulator'} log exited with code '${code}'`;
-        log.info(msg);
-        if (code) {
-          return reject(msg);
+      return (async () => {
+        try {
+          let startTime = process.hrtime();
+          await this.xcodebuild.start();
+          await this.waitForStart(startTime);
+          resolve();
+        } catch (err) {
+          let msg = `Unable to start WebDriverAgent: ${err}`;
+          log.error(msg);
+          reject(new Error(msg));
+          return;
```

Comment:
Not a whole lot. There must have been something after the catch block at one point.